contracts/token/ERC20Rebasable.sol

```diff
@@ -0,0 +1,352 @@
+// SPDX-FileCopyrightText: 2022 Lido <info@lido.fi>
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity 0.8.10;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {IERC20Wrapable} from "./interfaces/IERC20Wrapable.sol";
+import {IERC20BridgedShares} from "./interfaces/IERC20BridgedShares.sol";
+import {ITokenRateOracle} from "./interfaces/ITokenRateOracle.sol";
+import {ERC20Metadata} from "./ERC20Metadata.sol";
+
+/// @author kovalgek
+/// @notice Extends the ERC20Shared functionality
+contract ERC20Rebasable is IERC20, IERC20Wrapable, IERC20BridgedShares, ERC20Metadata {
+
+    error ErrorZeroSharesWrap();
+    error ErrorZeroTokensUnwrap();
+    error ErrorInvalidRateDecimals(uint8);
+    error ErrorWrongOracleUpdateTime();
+    error ErrorOracleAnswerIsNegative();
+    error ErrorTrasferToRebasableContract();
+    error ErrorNotEnoughBalance();
+    error ErrorNotEnoughAllowance();
+    error ErrorAccountIsZeroAddress();
+    error ErrorDecreasedAllowanceBelowZero();
+    error ErrorNotBridge();
+    error ErrorERC20Transfer();
+
+    /// @inheritdoc IERC20BridgedShares
+    address public immutable bridge;
+
+    /// @notice Contract of non-rebasable token to wrap.
+    IERC20 public immutable wrappedToken;
+
+    /// @notice Oracle contract used to get token rate for wrapping/unwrapping tokens.
+    ITokenRateOracle public immutable tokenRateOracle;
+    
+    /// @inheritdoc IERC20
+    mapping(address => mapping(address => uint256)) public allowance;
+
+    /// @notice Basic unit representing the token holder's share in the total amount of ether controlled by the protocol.
```

Comment:
```suggestion
    /// @notice Amount of shares (locked wstETH amount) owned by the holder
```
I think these comments should be closer to the implementation. Also it is not accurate from the L2 point of view